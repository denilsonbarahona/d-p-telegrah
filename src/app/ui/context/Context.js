import React from "react";
import initialState from "@context/initState";
import PropTypes from "prop-types";

const Context = React.createContext(null);

export const ContextProvider = ({ children }) => {
  return <Context.Provider value={initialState()}>{children}</Context.Provider>;
};

ContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Context;
