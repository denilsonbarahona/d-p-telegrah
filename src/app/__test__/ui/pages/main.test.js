import React from "react";
import { render } from "@testing-library/react";
import { ContextProvider } from "@context/Context";
import Main from "@pages/main";
import RouterMock from "../../../__mocks__/routes/routeMock";
import ProviderMock from "../../../__mocks__/redux/providerMock";


describe("main pages", ()=>{
    it("rendering without crashing", ()=>{
        const { container } = render(
            <ContextProvider>
                <ProviderMock>
                    <RouterMock>
                        <Main />
                    </RouterMock>
                </ProviderMock>
            </ContextProvider>);
        expect(container).toMatchSnapshot();
    });
});


