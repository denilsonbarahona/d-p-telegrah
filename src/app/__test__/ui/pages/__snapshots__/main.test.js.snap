// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main pages rendering without crashing 1`] = `
<div>
  <div
    class="sc-fnykZs iFSEpp"
  >
    <aside
      aria-label="side menu"
      class="sc-fEOsli hmhnzY"
    />
    <h2>
      placeholder
    </h2>
  </div>
</div>
`;
